library/common/extensions/filters/http/platform_bridge/filter.h

```diff
@@ -55,15 +55,23 @@ class PlatformBridgeFilter final : public Http::PassThroughFilter,
   Http::FilterTrailersStatus encodeTrailers(Http::ResponseTrailerMap& trailers) override;
 
 private:
+  static void replaceHeaders(Http::HeaderMap& headers, envoy_headers c_headers);
   Http::FilterHeadersStatus onHeaders(Http::HeaderMap& headers, bool end_stream,
                                       envoy_filter_on_headers_f on_headers);
   Http::FilterDataStatus onData(Buffer::Instance& data, bool end_stream,
-                                Buffer::Instance* internal_buffer, envoy_filter_on_data_f on_data);
+                                Buffer::Instance* internal_buffer,
+                                Http::HeaderMap** pending_headers, envoy_filter_on_data_f on_data);
   Http::FilterTrailersStatus onTrailers(Http::HeaderMap& trailers,
+                                        Buffer::Instance* internal_buffer,
+                                        Http::HeaderMap** pending_headers,
                                         envoy_filter_on_trailers_f on_trailers);
   const std::string filter_name_;
   IterationState iteration_state_;
   envoy_http_filter platform_filter_;
+  Http::HeaderMap* pending_request_headers_{};
+  Http::HeaderMap* pending_response_headers_{};
+  Http::HeaderMap* pending_request_trailers_{};
+  Http::HeaderMap* pending_response_trailers_{};
```

Comment:
I actually don't see that these references are being cleared - are we assuming they'll be automatically released when the filter is deallocated and that's sufficient?

Reply:
These member pointers are all being initialized to `nullptr`. This syntax is idiomatic (though it's also newer and perhaps less obvious to an unfamiliar observer than ` = nullptr;`). When they're set to something else in the implementation, they're being set to point to an object they don't manage - something else owns and will clean up that object. (This may be inferred by the type signatures that provide those objects.) The actual space the pointer itself occupies will be automatically reclaimed when the object is destructed.

(Don't worry - these details and their syntax are not all that intuitive.)